lib/hooks/useUppyUploader.js

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+
+import Uppy from "@uppy/core";
+import { useUppy } from "@uppy/react";
+import { Toastr } from "neetoui";
+
+import { UPPY_UPLOAD_CONFIG } from "components/Editor/MediaUploader/constants";
+import { convertToFileSize } from "components/Editor/MediaUploader/utils";
+import ActiveStorageUpload from "utils/ActiveStorageUpload";
+
+const useUppyUploader = ({
+  endpoint,
+  uppyConfig,
+  uppyUploadConfig = UPPY_UPLOAD_CONFIG,
+  onSuccess = _ => {},
+}) => {
+  const [isUploading, setIsUploading] = useState(false);
+  const uppy = useUppy(() =>
+    new Uppy({
+      ...uppyConfig,
+      onBeforeFileAdded: file => {
+        const { maxFileSize, allowedFileTypes } = uppyConfig.restrictions;
+
+        if (file.size > maxFileSize) {
+          Toastr.error(
+            `File size is too large. Max size is  ${convertToFileSize(
+              uppyConfig.restrictions.maxFileSize
+            )}.`
+          );
+
+          return false;
+        } else if (!allowedFileTypes.includes(`.${file.extension}`)) {
+          Toastr.error(
+            `File type is not permitted. Allowed file types are: ${allowedFileTypes.join(
+              ", "
+            )}.`
+          );
+
+          return false;
+        }
+
+        return true;
+      },
+    })
```

Comment:
Please extract this to a function named `handleBeforeAdded` and pass it here. This will improve the readability of this hook.